daisy_workflows/image_import/import_image.sh

```diff
@@ -139,14 +139,17 @@ fi
 
 # Ensure the output disk has sufficient space to accept the disk image.
 # Disk image size info.
-SIZE_BYTES=$(qemu-img info --output "json" "${IMAGE_PATH}" | grep -m1 "virtual-size" | grep -o '[0-9]\+')
+FILE_INFO=$(qemu-img info --output "json" "${IMAGE_PATH}")
+SIZE_BYTES=$(echo $FILE_INFO | grep -oP  '(?<="virtual-size":[ *])[0-9]+' | head -1)
+IMPORT_FILE_FORMAT=$(echo $FILE_INFO | grep -oP  '(?<="format":[ *]").*?(?=")')
```

Comment:
Have you tested `qemu-img info` with various formats? Is it reliable? Does it happen that format is not present in the JSON? We should handle that scenario in case it ever happens, so we don't break the import just because format is not present. I would test this grep statement with a JSON without "format" field. If that happens, you can fallback to file extensions of the imported file.

Reply:
yes it is stable.
Even if failed to fetch the format, it won't break anything, just got an empty format.

When matching failed (which won't happen per current testing), I would leave it empty other than file extension because file extension is not reliable and in many cases it can be empty.